### PR TITLE
 [FEAT]: Send,edit, delete handler

### PR DESCRIPTION
**Pull request by VipinDevelops**
## Issue(s)

#5 

## Acceptance Criteria fulfillment
This PR implements the functionality for sending, editing, and deleting replies using a block action handler.

- [x]  **Send Handler and Modal**:

   - Developed a handler to process the send action.
   - Designed and implemented a modal for sending replies.
   
- [x] **Edit Handler and Modal**:
   - Developed a handler to process the edit action.
   - Designed and implemented a modal for editing replies.
   
- [x] **Delete Handler and Modal**:
   - Developed a handler to process the delete action.
   - Designed and implemented a modal for deleting replies.

## Proposed changes (including videos or screenshots)

[Screencast from 2024-06-25 02-35-21.webm](https://github.com/RocketChat/Apps.QuickReplies/assets/99081689/b93d57c3-1ef9-46c2-a59f-6fddfe8e638c)

### CODE REVIEW

File: src/enum/modals/createModal.ts [hunk 1]
```diff
@@ -1,4 +1,4 @@
-export enum CreateModalEnum {
+export enum createModalEnum {
```
Comment (hugocostadev) [inline src/enum/modals/createModal.ts:1]: Is there a reason why you changed the case of the enums? I'd suggest keeping the CamelCase

File: src/handlers/ExecuteViewSubmitHandler.ts [hunk 2]
```diff
@@ -150,15 +177,196 @@ export class ExecuteViewSubmitHandler {
...
+		const result = await replyStorage.deleteReplyById(
+			user,
+			replyId,
+			language,
+		);
+
+		if (result.success) {
+			const successMessage = `${t('quick_reply', language)} **${
+				cachedReply.name
+			}** ${t('delete_successfully', language)}✅`;
```
Comment (hugocostadev) [inline src/handlers/ExecuteViewSubmitHandler.ts:253]: This happens in other places too, so consider changing in all occurrences... 

When you want to display a composable message, don't "merge" i18n keys, some languages have different orders of sentences

I'd suggest using variables interpolations https://react.i18next.com/latest/trans-component#interpolation

Comment (hugocostadev) [reply src/handlers/ExecuteViewSubmitHandler.ts:253]: You can also use the <Trans> component

File: src/handlers/ExecuteViewSubmitHandler.ts [hunk 3]
```diff
@@ -150,15 +177,196 @@ export class ExecuteViewSubmitHandler {
...
+			persistenceRead,
+		);
+
+		const replyCacheStorage = new CacheReplyStorage(
+			this.persistence,
+			this.read.getPersistenceReader(),
+		);
+
+		let name = '';
+		let body = '';
```
Comment (hugocostadev) [inline src/handlers/ExecuteViewSubmitHandler.ts:304]: replace with const and move to the initialization (line 319)

**Review by hugocostadev — CHANGES_REQUESTED** [review]

File: src/enum/modals/createModal.ts (hunk 1 above)
Comment (VipinDevelops) [reply src/enum/modals/createModal.ts:1]: Sure, I was just confused which one should we use

File: i18n/en.json
[the remark this answers is not in the document]
Comment (VipinDevelops) [reply i18n/en.json:1]: Hey @hugocostadev  can you explain this with a small example I didn't get it.

File: src/handlers/ExecuteViewSubmitHandler.ts (hunk 2 above)
Comment (VipinDevelops) [reply src/handlers/ExecuteViewSubmitHandler.ts:253]: we can also use the` __parameter__ `to insert the parameters

**Comment by VipinDevelops** [comment]
Hey @hugocostadev  I made the requested changes.

**Pull request merged**